fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -2107,19 +2107,37 @@ private void createOlapOrLakeTable(Database db, CreateTableStmt stmt) throws Ddl
         if (stmt.isExternal()) {
             table = new ExternalOlapTable(db.getId(), tableId, tableName, baseSchema, keysType, partitionInfo,
                     distributionInfo, indexes, properties);
-        } else {
+        } else if (stmt.isOlapEngine()) {
             if (distributionInfo.getBucketNum() == 0 || Config.enable_auto_tablet_distribution) {
                 int bucketNum = CatalogUtils.calBucketNumAccordingToBackends();
                 distributionInfo.setBucketNum(bucketNum);
             }
 
-            if (stmt.isOlapEngine() && RunMode.getCurrentRunMode().isAllowCreateLakeTable()) {
+            RunMode runMode = RunMode.getCurrentRunMode();
+            String volume = stmt.getProperties().get(PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME);
```

Comment:
better add a function to get storage volume in `PropertyAnalyzer`
and remove it from properties because properties empty check will do later.

Reply:
I'll submit another patch to fix this issue.